core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -78,7 +78,11 @@ object UniqueTriggerActivation {
                 return true
             }
             OneTimeFreeUnitRuins -> {
-                val unit = civInfo.getEquivalentUnit(unique.params[0])
+                var unit = civInfo.getEquivalentUnit(unique.params[0])
+                if ( unit.hasUnique(UniqueType.FoundCity) && civInfo.isOneCityChallenger()) {
+                    unit = civInfo.getEquivalentUnit(Constants.worker)
```

Comment:
Remember that in modded games they may not have a worker unit! Or they could have one with a different name!